src/util/logger.js

```diff
@@ -8,6 +8,13 @@ const DailyRotateFile = require('winston-daily-rotate-file');
 winston.configure({
   exitOnError: false,
   transports: [
+    new (winston.transports.Console)({
+      colorize: true,
+      handleExceptions: true,
+      humanReadableUnhandledException: true,
+      level: 'error',
+      showLevel: true,
+    }),
```

Comment:
Is this different from `console.error`? Should we remove that?

Reply:
Do you mean from this line:

https://github.com/MarkBind/markbind/blob/1686253401a4ad6ef75f67dfab0e7adc5c93543e/src/lib/markbind/src/parser.js#L82

Reply:
No, this line:

https://github.com/MarkBind/markbind/blob/669325bcdefdb2b2d6ddb37b2ef6500e64555eac/src/util/logger.js#L33

Reply:
The current mechanism of logging is as follows: 

We are using `logger.js` functions to print messages to the console with `console.log()` statements and a [winston-daily-rotate-file transport](https://github.com/winstonjs/winston-daily-rotate-file) set to level `debug` that writes messages to a log file. 

Nunjucks errors were not being logged to the console because we are not calling `logger.error()` explicitly in `nunjucks.renderString()` (as we don't have error handlers for these calls). Other errors do get printed to the console because we call `logger.error()` when we catch them.

However, nunjucks errors were being printed to the log file because of the winston-daily-rotate-file transport. 

If we want to remove `console.error` (and other `console.log` statements in `logger.js`), then we should a winston console transport with the level set to `debug`. There is an option to use custom formats (color, etc.) too.

Reply:
Correct me if I am wrong, `nunjucks` is actually using `winston-daily-rotate-file` internally?

Reply:
No, `nunjucks` itself doesn't use `winston-daily-rotate-file`. `winston-daily-rotate-file` is a storage device for logs (i.e. a transport) where all our logging output goes. Currently MarkBind only uses this transport. To display logs on the console **with winston** we should have added a Console transport to our winston configuration.

The reason any log messages are being shown on the console right now is because `logger.js` explicitly outputs them using `console.log` statements.

Reply:
Hope this makes sense!

Reply:
1. Where do we configure winston for nunjucks?

2. > If we want to remove `console.error` (and other `console.log` statements in `logger.js`), then we should use the winston console transport with the level set to `debug`.

   Why is level set to `error` insufficient to remove `console.error`?

Reply:
> 1. Where do we configure winston for nunjucks?

If I'm not wrong, we are configuring winston only once in `logger.js` and using the logger instance in the rest of the application.

> 2. Why is level set to `error` insufficient to remove `console.error`?

For removing `console.error`, setting the level to `error` is sufficient. If we want to take it one step further and remove statements like `console.log(chalk.cyan('info: ') + text)` (i.e. for other levels too), `debug` would work.

Reply:
1. So how does nunjucks use winston?

2. As asked in my first comment: *Should we remove that?*

Reply:
> 1. So how does nunjucks use winston?

Our application as a whole uses winston, so all logging messages are handled by it. We don't need to configure nunjucks to use it, any message / uncaught exception from nunjucks will be logged automatically to _the log file_ (and only the log file, because that's the transport we have). 

> 2. As asked in my first comment: _Should we remove that?_

No we should not remove it, because we are calling `logger.error()` from other parts of the code. For example, our errorHandler uses it:

https://github.com/MarkBind/markbind/blob/6bbe1fa7d8a5eb02ff160050bc1fc088b9808f2e/src/Page.js#L766

Reply:
1. How does our application use winston if not through logger?

2. Logger calls winston.

Reply:
Our application does use winston through the logger, but we have also configured winston to handle logging uncaught exceptions. 
https://github.com/MarkBind/markbind/blob/6bbe1fa7d8a5eb02ff160050bc1fc088b9808f2e/src/util/logger.js#L15
That's how nunjucks errors were being logged to the log file.

Reply:
1. Thanks for checking this. It answers the question originally raised by @yamgent.

2. How about this?

Reply:
> 1. Let's set `winston.transports.Console` level to `'info'` and then remove `console.log` in `logger.error`, `logger.warn` and `logger.info`.

May I do these changes in this PR itself?

Reply:
Yes please.